src/python/WMCore/Services/WMStats/WMStatsWriter.py

```diff
@@ -77,6 +78,27 @@ def uploadData(self, docs):
             self.couchDB.queue(doc)
         return self.couchDB.commit(returndocs=True)
 
+    def bulkUpdateData(self, docs, existingDocs):
+        """
+        :param docs: docs to insert or update
+        :param existingDocs: docs existing in current
+        :return:
+        """
+        if isinstance(docs, dict):
+            docs = [docs]
+        for doc in docs:
+            if doc['_id'] in existingDocs:
+                revList = existingDocs[doc['_id']].split('-')
+                # update the revision number
+                doc['_rev'] = "%s-%s" % (int(revList[0]) + 1, revList[1])
+            else:
+                # just send well formatted revision for the new documents which required by new_edits=False
+                doc['_rev'] = "1-123456789"
```

Comment:
Same question here. If document doesn't exist, don't we just post it and let the revision to magically appear?

Reply:
The same answer. We can separate the documents with initial inserting and updating then use new_edits=True, new_edits=False respectively but I don't see any advantages doing that.